src/game/client/draw/w3dmodeldraw.h

```diff
@@ -0,0 +1,390 @@
+{
+    struct HideShowSubObjInfo
+    {
+        Utf8String name;
+        bool visible;
+    };
+
+    struct TurretInfo
+    {
+        void Clear()
+        {
+            m_turretAngleName = NAMEKEY_INVALID;
+            m_turretPitchName = NAMEKEY_INVALID;
+            m_turretArtAngle = 0.0f;
+            m_turretArtPitch = 0.0f;
+            m_turretAngleBone = 0;
+            m_turretPitchBone = 0;
+        }
+
+        NameKeyType m_turretAngleName;
+        NameKeyType m_turretPitchName;
+        float m_turretArtAngle;
+        float m_turretArtPitch;
+        int m_turretAngleBone;
+        int m_turretPitchBone;
+    };
+
+    struct WeaponBarrelInfo
+    {
+        WeaponBarrelInfo() { Clear(); }
+
+        void Set_Muzzle_Flash_Hidden(RenderObjClass *robj, bool hidden) const;
+
+        void Clear()
+        {
+            m_weaponRecoilBone = 0;
+            m_weaponFireFXBone = 0;
+            m_weaponMuzzleFlashBone = 0;
+            m_weaponLaunchBoneTransform.Make_Identity();
+            m_weaponMuzzleFlashBoneName.Clear();
+        }
+
+        int m_weaponRecoilBone;
+        int m_weaponFireFXBone;
+        int m_weaponMuzzleFlashBone;
+        Matrix3D m_weaponLaunchBoneTransform;
+        Utf8String m_weaponMuzzleFlashBoneName;
+    };
+
+    ModelConditionInfo() { Clear(); }
+    void Add_Public_Bone(Utf8String const &bone) const;
+    void Clear();
+    Matrix3D const *Find_Pristine_Bone(NameKeyType key, int *index) const;
+    bool Find_Pristine_Bone_Pos(NameKeyType key, Coord3D &pos) const;
+    void Load_Animations() const {}
+    bool Matches_Mode(bool night, bool snow) const;
+    void Preload_Assets(TimeOfDayType time_of_day, float scale);
+    void Validate_Cached_Bones(RenderObjClass *robj, float scale) const;
+    void Validate_Stuff(RenderObjClass *robj, float scale, std::vector<Utf8String> const &bones) const;
+    void Validate_Turret_Info() const;
+    void Validate_Weapon_Barrel_Info() const;
+
+    const BitFlags<MODELCONDITION_COUNT> &Get_Conditions_Yes(int condition_idx) const
+    {
+        return m_conditionsYesVec[condition_idx];
+    }
+
+    int Get_Conditions_Count() const { return m_conditionsYesVec.size(); }
+    Utf8String Get_Definition() const { return m_description; }
+
+    Utf8String m_description;
+    std::vector<BitFlags<MODELCONDITION_COUNT>> m_conditionsYesVec;
+    Utf8String m_modelName;
+    std::vector<HideShowSubObjInfo> m_hideShowVec;
+    std::vector<Utf8String> m_publicBones;
+    Utf8String m_weaponFireFXBoneName[3];
+    Utf8String m_weaponRecoilBoneName[3];
+    Utf8String m_weaponMuzzleFlashName[3];
+    Utf8String m_weaponLaunchBoneName[3];
+    Utf8String m_weaponHideShowBoneName[3];
+    std::vector<W3DAnimationInfo> m_animations;
+    NameKeyType m_transitionKey;
+    NameKeyType m_allowToFinishKey;
+    int m_flags;
+    int m_iniReadFlags;
+    int m_mode;
+    std::vector<ParticleSysBoneInfo> m_ParticleSysBones;
+    uint64_t m_transition;

```

Comment:
don't see this anywhere....

Reply:
Look at 00620499 in wb, the function 00633260 returns a uint64_t that is then put into m_transition